engine/src/main/java/org/terasology/rendering/dag/nodes/RefractiveReflectiveBlocksNode.java

```diff
@@ -278,39 +278,40 @@ public void process() {
-    public void update() {
-        lastUpdatedGBuffer.attachDepthBufferTo(refractiveReflectiveFbo);
-    }
-
-    @Override
     public void propertyChange(PropertyChangeEvent event) {
-        // This method is only called when oldValue != newValue.
-        if (event.getPropertyName().equals(RenderingConfig.NORMAL_MAPPING)) {
-            normalMappingIsEnabled = renderingConfig.isNormalMapping();
-            if (normalMappingIsEnabled) {
-                addDesiredStateChange(setTerrainNormalsInputTexture);
-                if (parallaxMappingIsEnabled) {
-                    addDesiredStateChange(setTerrainHeightInputTexture);
-                }
-            } else {
-                removeDesiredStateChange(setTerrainNormalsInputTexture);
-                if (parallaxMappingIsEnabled) {
-                    removeDesiredStateChange(setTerrainHeightInputTexture);
-                }
-            }
-        } else if (event.getPropertyName().equals(RenderingConfig.PARALLAX_MAPPING)) {
-            parallaxMappingIsEnabled = renderingConfig.isParallaxMapping();
-            if (normalMappingIsEnabled) {
-                if (parallaxMappingIsEnabled) {
-                    addDesiredStateChange(setTerrainHeightInputTexture);
+        if (event.getPropertyName().equals(DisplayResolutionDependentFBOs.PRE_FBO_REGENERATION)) {
+            refractiveReflectiveFbo.detachDepthBuffer();
+        } else if (event.getPropertyName().equals(DisplayResolutionDependentFBOs.POST_FBO_REGENERATION)) {
+            lastUpdatedGBuffer.attachDepthBufferTo(refractiveReflectiveFbo);
+        } else { // Else: One of the settings was changed, handle the event and refresh the task list.
+            // This method is only called when oldValue != newValue.
+            if (event.getPropertyName().equals(RenderingConfig.NORMAL_MAPPING)) {
+                normalMappingIsEnabled = renderingConfig.isNormalMapping();
+                if (normalMappingIsEnabled) {
+                    addDesiredStateChange(setTerrainNormalsInputTexture);
+                    if (parallaxMappingIsEnabled) {
+                        addDesiredStateChange(setTerrainHeightInputTexture);
+                    }
                 } else {
-                    removeDesiredStateChange(setTerrainHeightInputTexture);
+                    removeDesiredStateChange(setTerrainNormalsInputTexture);
+                    if (parallaxMappingIsEnabled) {
+                        removeDesiredStateChange(setTerrainHeightInputTexture);
+                    }
                 }
-            }
-        } else if (event.getPropertyName().equals(RenderingConfig.ANIMATE_WATER)) {
-            animatedWaterIsEnabled = renderingConfig.isAnimateWater();
-        } // else: no other cases are possible - see subscribe operations in initialize().
+            } else if (event.getPropertyName().equals(RenderingConfig.PARALLAX_MAPPING)) {
+                parallaxMappingIsEnabled = renderingConfig.isParallaxMapping();
```

Comment:
This is an issue inherited from the previous version, but I think this line can go one line below, after the `if(normalMappingIsEnabled)` line.

Reply:
Nice catch :D